test/unified_malloc_framework/memoryPool.hpp

```diff
@@ -251,6 +276,46 @@ TEST_P(umfPoolTest, multiThreadedMallocFreeRandomSizes) {
     }
 }
 
+TEST_P(umfMemTest, outOfMem) {
+    static constexpr size_t allocSize = 4096;
+    auto hPool = pool.get();
+
+    std::vector<void *> allocations;
+
+    while (true) {
+        allocations.emplace_back(umfPoolMalloc(hPool, allocSize));
+        if (allocations.back() == nullptr &&
+            umfPoolGetLastAllocationError(hPool) ==
+                UMF_RESULT_ERROR_OUT_OF_HOST_MEMORY) {
+            break;
+        }
+        ASSERT_NE(allocations.back(), nullptr);
+    }
+
+    // next part of the test- freeing some memory to allocate it again (as the memory
+    // should be acquired from the pool itself now, not from the provider),
+    // is done only for the disjoint pool for now
+
+    // remove last nullptr from the allocations vector
+    ASSERT_EQ(allocations.back(), nullptr);
+    allocations.pop_back();
+
+    ASSERT_NE(allocations.back(), nullptr);
```

Comment:
I'm not exactly sure what you mean. Could you clarify? The size of the `allocations` vector is larger than the value of `expectedPooledAllocations`, since the latter is used to free some memory and then for trying to allocate it again (to check that memory is taken now  from the pool, not from the provider).

Reply:
I mistook `expectedPooledAllocation` for the number of allocations from the pool before it goes OOM. In that case maybe `expectedRecycledPoolAllocs` would fit better.

Reply:
Done.